### ISSUE TITLE
Foundation/SASS branch proposal

### ISSUE THREAD

**Issue by boxxxie** [ngbp/ngbp#191]
Would this be of interest to you guys?

I am currently doing this for the project i'm working on right now, and so far I have replaced the recess in grunt to use the sass plugin and i have it compiling the foundation scss files.

the HTML needs to be changed to work with foundation (willing to do that as well since it's only 2 pages of html).

The installer would have to be edited to install sass and maybe ruby.

**Comment by richardwardza** [comment]
There hasn't been much action on it lately but check out https://github.com/MikeFielden/ng-compass-boilerplate

**Comment by boxxxie** [comment]
well. i guess i'll do that as well.
I didn't know about other people working on SASS forks until this issue.

maybe it would be a good idea to include a reference to ng-compass-boilerplate in the readme, as it would have saved me a bit of time if i had known about it.

**Comment by richardwardza** [comment]
It's not up to date with the current ng-boilerplate though! (last activity was a couple of months ago as far as I know)

**Comment by boxxxie** [comment]
the edits in the gruntfile would have been good enough for me.

i made very minor changes to switch to foundation.

**Comment by joshdmiller** [comment]
Good news! The upcoming version of ngbp will support sass or of the box. It's a significant overhaul and I'll be asking for beta testers next week; anyone interested?

**Comment by boxxxie** [comment]
sounds good to me. i can try it with foundation.

**Comment by richardwardza** [comment]
I can give it a whirl with a new app I'm wanting to do..

**Comment by saamalik** [comment]
@joshdmiller likewise. I'm interested in testing with SASS support as well.

**Comment by rosslavery** [comment]
@joshdmiller I'd be very interested in testing out the new version.

**Comment by eladmoshe** [comment]
@joshdmiller I'm also interested.

**Comment by guyroutledge** [comment]
I'm looking to start a new Angular project soon will be using Sass/Compass rather than Less. I very much like the structure provided by ng-boilerplate. Is a Sass release still in the pipeline?

**Comment by joshdmiller** [comment]
@guyroutledge Yep, for sure. I'm still working out the details; if you're curious what's in the next release, see this: https://github.com/ngbp/ng-boilerplate/issues/173#issuecomment-27824949.

Just a small clarification: it won'tbe a "sass release" in so much as it will support sass and less both out of the box, with neither enabled by default. You can enable sass by running `grunt ngbp:use:ngbp-contrib-sass`.

**Comment by kylegirard** [comment]
@joshdmiller Also, very interested to test the new version.

**Comment by meenie** [comment]
@joshdmiller, totally interested in the beta!! :)

**Comment by joshdmiller** [comment]
Awesome. How would you guys feel about a little Google+ hangout maybe at the end of next week with whomever is interested in beta testing 0.4. I don't have documentation up yet (I anticipate some significant changes prior to the release) so I will show everyone how to use it, answer questions, etc.

**Comment by meenie** [comment]
@joshdmiller - Sounds good!

**Comment by lezhangxyz** [comment]
I'd love to test out the new version as well. Two production projects are heavily invested in ngbp :)

**Comment by pjackowski** [comment]
Thanks, you are AW3S0M3! ;-) When do you expect new release? In  mean time I will definitely look into ng-compass-boilerplate!

I have spent couple hours trying to make this thing working with SASS or try to use component oriented folder structure together with different Angular generator, but I really had difficult time. I can test possibly something, if I only know what to do. I relatively new to SASS and Angular.

**Comment by joshdmiller** [comment]
Anyone still interested in helping out with v0.4, sorry for the delay. December was hugely busy for me, but development has continued regardless! In fact, v0.4 [already has SASS support build in](https://github.com/ngbp/ngbp-contrib-sass) and there will be a Foundation example repository to show it off.

We are finishing up a few things before doing the hangout I mentioned before; I'm hoping end of next week. If anyone is interested in helping out _developing_ the final features of v0.4, shoot me an email - there's plenty to do!

**Comment by rdthompson** [comment]
I'd be glad to help out as soon as my current AngularJS gig is up.

I think this is a great project.

**Comment by th3morg** [comment]
@joshdmiller I know its been a while, but if you are still looking for beta testers, I'm actively using 0.3.1 and would be happy to try out 0.4!

**Comment by rdthompson** [comment]
Can someone guess whether Angular and other tools are going to end up a massive mess of text tools and compiling.

Certainly this is a huge paradigm shift using this like, I know, ng-repeat to replace the traditional for, for each tool.

Yes it does have that, but this is a different way of developing that requires the Html5 SEMANTICS to be really tight and agreed upon and that's not what's happening.

Instead the number of frameworks and tools it OUTPACING THE HTML5 language.

In fact Angular is almost a replacement for Html5.

Is everyone comfortable with this?  I've seen the growth of WWW since 1990 when I was first asked to join the WWW project in 1990.

It's gone through so many twists and turns its mindboggling.

~ Robert

**Comment by josephdpurcell** [comment]
I'm taking a look at v0.4 and I hope I'm not being dense here, but (compared to v0.3) it is very empty. For example, I don't see a gruntfile or bower.json and the README is empty.

The setup process I did for v0.3 was:
npm install
bower install
grunt
and then everything worked fine. Is there a similar process for v0.4?

**Comment by joshdmiller** [comment]
@josephdpurcell That branch was junk; I apologize for that. I cleared out all the irrelevant/outdated branches to avoid confusion.

**Attention all who want to try with SASS:** I have an alpha-level preview for you guys to check out. Whoever's interested, see http://getwarlock.com and check out the instructions. I'm really hoping for some testers who will check things out and ask great questions. There's no documentation to speak of, so I expect quite a few.

**Comment by julkue** [comment]
Is there anything new about SCSS/SASS instead of LESS? I found different projects that have implemented SCSS instead of LESS but unfortunately they are several years old and I'm not in a position to use old software...

**Comment by julkue** [comment]
@joshdmiller

**Comment by julkue** [comment]
Since there was no reply and no interest in implementing SCSS/SASS I've created another [boilerplate](https://github.com/julmot/jmBoilerplate). Just for those who are interested in a framework like ngbp but with these improvements.
